src/kohlrahbi/docxfilefinder.py

```diff
@@ -11,6 +12,136 @@
 from kohlrahbi.logger import logger
 
 
+class EdiEnergyDocument(BaseModel):
+    """
+    This class represents an EDI Energy document.
+    """
+
+    filename: Path
+    document_version: str
+    version_major: int
+    version_minor: int
+    version_suffix: str
+    valid_from: int
+    valid_until: int
+
+    @classmethod
+    def from_path(cls, path: Path) -> "EdiEnergyDocument":
+        """
+        Create an EdiEnergyDocument object from a file path.
+        """
+
+        file_name = extract_document_version_and_valid_dates(path.name)
+        assert file_name is not None, f"Could not extract document version and valid dates from {path.name}."
+        return cls(
+            filename=path,
+            document_version=file_name["document_version"],
+            version_major=int(file_name["version_major"]),
+            version_minor=int(file_name["version_minor"]),
+            version_suffix=file_name["version_suffix"],
+            valid_from=int(file_name["valid_from"]),
+            valid_until=int(file_name["valid_until"]),
+        )
+
+    def __lt__(self, other: "EdiEnergyDocument") -> bool:
+        """
+        Compare two EdiEnergyDocument instances based on
+        their document_version(major, minor and suffix), valid_until, and valid_from.
+
+        I did not know how the tuple comparison works in Python, so I looked it up:
+
+        Python compares tuples lexicographically, meaning it compares the elements one by one from left to right.
+        The comparison starts with the first elements of both tuples:
+          If self.valid_from is less than other.valid_from, the entire expression evaluates to True.
+          If self.valid_from is greater than other.valid_from, the entire expression evaluates to False.
+          If self.valid_from is equal to other.valid_from, Python moves to the next elements in the tuples.
+        This process continues with self.valid_until vs. other.valid_until and then with the version numbers.
+
+        Args:
+            other (EdiEnergyDocument): The other document to compare against.
+
+        Returns:
+            bool: True if this document is considered less than the other document, False otherwise.
+        """
+        is_less_than = (self.valid_from, self.valid_until, self.version_major, self.version_minor) < (
```

Comment:
krass, müssen wir erst die Daten vergleichen und am Ende erst die Versionsnummer?

Reply:
In den meisten Fällen ist die Versionsnummer gleich pro Formatversion (verschiedene Fehlerkorrekturen) und in seltenen Fällen gibt es zu den selben Daten verschiedene Versionsnummern. Aber im Prinzip ist die Reihenfolge egal. Das Versions Suffix (z.B. "a" in 2.1a) wird nur relevant, wenn alles andere gleich ist, weil sonst der Vergleich bricht (z.B: 3.1a vs 2.1b). Oh, da fällt mir gerade noch ein kleiner Bug auf 🙈

Reply:
93264fb983a5a461bc30095b4945fe0ab841d6f5